presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/LinuxMemoryChecker.h"
+#ifdef __linux__
+#include <boost/regex.hpp>
+#include <folly/File.h>
+#include <folly/FileUtil.h>
+#include <folly/gen/Base.h>
+#include <folly/gen/File.h>
+#include <folly/gen/String.h>
+
+#include <folly/Conv.h>
+#include <folly/CppAttributes.h>
+#include <folly/Format.h>
+#include <folly/Range.h>
+#include <folly/String.h>
+
+#include <sys/stat.h>
+
+#endif // __linux__
+
+namespace facebook::presto {
+
+// Current memory calculation used is inactive_anon + active_anon
```

Comment:
I don't think LinuxMemoryChecker should be added to https://prestodb.io/docs/current/presto_cpp/properties.html since it's not a coordinator or worker property that can be set by the .properties file, it can only be enabled in CMakeLists.txt

Reply:
https://prestodb.io/docs/current/presto_cpp/properties.html has the topics _Coordinator Properties_ and _Worker Properties_. 

I would think it reasonable to add to that page a third topic named (something like) _Build-enabled Properties_ listing these properties, with a short intro paragraph explaining these are only present if they were enabled at build time by (explanation involving CMakeLists.txt).

Reply:
+1 to both Christian and Steve's suggestions.

Other than the config, I feel our experience with the Periodic memory checker as is documented in this file is a good candidate for a blog post. wdyt ?

Reply:
Created PR for doc, please review when you can! @steveburnett @aditi-pandit @czentgr 
https://github.com/prestodb/presto/pull/23460

Reply:
Merged doc PR into this PR